src/cc/BPFTable.h

```diff
@@ -143,4 +250,41 @@ class BPFPerfBuffer : protected BPFTableBase<int, int> {
   std::unique_ptr<epoll_event[]> ep_events_;
 };
 
+class BPFProgTable : protected BPFTableBase<int, int> {
+private:
+  // this class is used to allow updating a table element using the
+  // table[index] = value syntax
+  class BPFProgTableWriter {
```

Comment:
The prog array class is essentially the same as normal array. Do you think it would be a good idea to inherit and re-use code?